lib/runner.js

```diff
@@ -77,7 +77,16 @@ Object.assign(Runner.prototype, {
     PassThrough = PassThrough || require('stream').PassThrough;
 
     const runner = this;
-    const stream = new PassThrough({ objectMode: true });
+    let queryStream;
+    const stream = new PassThrough({
+      objectMode: true,
+      destroy() {
+        if (!queryStream) return;
+        queryStream.destroy();
+        queryStream.emit('error', new Error('stream destroyed'));
```

Comment:
```suggestion
        queryStream.destroy(new Error('stream destroyed'));
```

Based on https://nodejs.org/api/stream.html#stream_readable_destroy_error you can just pass `Error` to `destroy()`

Reply:
This should work, but not for `mysql2`. `queryStream` from `mysql2` is not initial readable stream and destroying  this stream does not destroy source stream, so for `mysql2` this still does not work. But this is `mysql2` [issue #1168](https://github.com/sidorares/node-mysql2/issues/1168).

But this PR change is correct. If `PassThrough` stream here is destroyed, then by default this does not destroy parent/source stream, so it must be destroyed separately, otherwise source stream will stay paused (because of unpipe). And it is up  to underlying dialect's client to also make sure that source stream is actually destroyed and query to database cancelled.